### PR TITLE
perf: vite plugin process css logic

### PR DESCRIPTION
**Pull request by nonzzz**
## Linked Issues

#94 

## Description

This pull request won't need option `fileName`. Perf `renderChunk` logic so that it can work better with vite.

### CODE REVIEW

File: vite.js [hunk 1]
```diff
@@ -82,19 +91,32 @@ module.exports = function style9Plugin(opts = {}) {
...
     },
-    renderChunk(_, chunk) {
+    async renderChunk(_, chunk) {
       const ids = Object.keys(chunk.modules);
+      // https://github.com/vitejs/vite/blob/main/packages/vite/src/node/plugins/css.ts#L482-L489
       for (const id of ids) {
         if (id.startsWith(VIRTUAL_CSS_NAME)) {
+          // fool the css plugin to generate the css in corresponding chunk
+          const fakeCssId = `${chunk.fileName}.css`;
+          if (cssPlugin && cssPluginPost) {
```
Comment (wmertens) [inline vite.js:101]: does this mean that if there's no cssplugin, the css is removed entirely?

**Comment by wmertens** [comment]
BTW, I think we can speed up processing a little bit by checking if the code contains `style9` before running babel on it

File: vite.js (hunk 1 above)
Comment (nonzzz) [reply vite.js:101]: Don't worry.vite contains this plugin

**Comment by nonzzz** [comment]
@wmertens I don't understand you mean. Should we need check it in `transform`?

**Comment by wmertens** [comment]
Yes correct, if style9 isn't mentioned there's no point in generating an AST

**Comment by nonzzz** [comment]
OK. I will try it

**Comment by wmertens** [comment]
And for the css plugin, maybe it's better to throw if the plugin isn't there?

**Comment by nonzzz** [comment]
@wmertens No. vite already includes these 2 plugins. This is also done in other similar libraries

**Comment by wmertens** [comment]
What I'm worried about is that the plugin only works if those two plugins are present, so IMHO either it should throw or it shouldn't test for them

**Comment by nonzzz** [comment]
Qwik is base on `vite` and vite contains `vite:css` and `vite:css-post` plugin. I will test Qwik in my local env.

**Comment by nonzzz** [comment]
You can look at [vite-css](https://github.com/vitejs/vite/blob/main/packages/vite/src/node/plugins/css.ts) and [vite-plugin](https://github.com/vitejs/vite/blob/main/packages/vite/src/node/plugins/index.ts#L29-L104)
If vite don't provide those plugin will panic

**Comment by nonzzz** [comment]
> @nonzzz when I add this as the first line of `transformStyle9`:
> 
> ```js
>   if (!/style9/.test(code)) return;
> ```
> 
> then the time spent goes from: Screenshot from 2023-05-15 09-04-16
> 
> to: image

cool.

File: vite.js [hunk 2]
```diff
@@ -81,9 +81,7 @@ module.exports = function style9Plugin(opts = {}) {
       return null;
     },
     transform(code, id) {
-      if (!filter(id)) {
-        return;
-      }
+      if (!filter(id) && !/style9/.test(code)) return;
```
Comment (wmertens) [inline vite.js:84]: this should be `||`

Comment (nonzzz) [reply vite.js:84]: why use `||`. If code don't have `style9` it will be skip. isn't it?

Comment (nonzzz) [reply vite.js]: my mistake.

**Comment by wmertens** [comment]
I can confirm that this change fixes the extra CSS file in Qwik for building, yey :-) 

In dev mode the style doesn't work, even after changing the file (which used to work because the hot reload had the correct css module path). Trying to find out why - it looks to be reloading the page instead of hot reloading.

I'm pretty sure this is a problem with Qwik though, so it shouldn't block this PR. Changing the ordering doesn't make a difference.

**Pull request closed without merging**